arbor/threading/threading.hpp

```diff
@@ -97,10 +97,27 @@ class task_system {
 
 class task_group {
 private:
+    struct exception_struct {
```

Comment:
The `_struct` is redundant, and could be replaced with something that describes what the struct is for, like `exception_state_`?
The members could also be simplified to
* `exception_ptr_` --> `exception_`
* `exception_mutext_` --> `mutex_`
* `bail` --> `error_`

Reply:
Thanks for the name suggestions and style comments. I have modified all the things you pointed out :)